index.rb

```diff
@@ -144,6 +146,38 @@ def show(**kwargs)
   end
 end
 
+venues << Venue.new(:name => 'Kilowatt', :link => 'https://kilowattbar.com/') do
+  URI.open(URI.join(link, 'events-1')) do |html|
+    # We need to pull the api key from a script tag inside an iframe. Nokogiri
+    # doesn't parse javascript, so a greasy regex will have to do for now.
+    Nokogiri::HTML(html).css('iframe').attr('srcdoc').text.match(/"apiKey":"(?<apikey>\w+)"/) do |match|
+      kw_api_key = match[:apikey]
+
+      if kw_api_key
+        uri = URI.parse('https://events-api.dice.fm/v1/events?page[size]=24&types=linkout,event&filter[venues][]=Kilowatt')
+        response = Net::HTTP.start(uri.host, uri.port, use_ssl: true) do |http|
+          request = Net::HTTP::Get.new(uri)
+          request['Accept'] = 'application/json'
+          request['x-api-key'] = kw_api_key
+
+          http.request(request)
+        end
+
+        if Net::HTTPOK === response
+          JSON.parse(response.body)['data'].map do |event|
```

Comment:
I like to use `fetch` so that errors are a little more helpful when something goes wrong. Same with `event` accesses below

Reply:
Good call. I sprinkled in some fetches for accessing keys in the parsed events so we will err early if the underlying format changes.